src/main/java/org/tikv/common/region/RegionStoreClient.java

```diff
@@ -449,15 +459,17 @@ public boolean onNotLeader(Store newStore) {
     if (logger.isDebugEnabled()) {
       logger.debug(region + ", new leader = " + newStore.getId());
     }
-    TiRegion cachedRegion = getSession().getRegionManager().getRegionById(region.getId());
+    TiRegion cachedRegion = regionManager.getRegionById(region.getId());
     // When switch leader fails or the region changed its key range,
     // it would be necessary to re-split task's key range for new region.
-    if (!region.switchPeer(newStore.getId())
-        || !region.getStartKey().equals(cachedRegion.getStartKey())
+    // region = region.switchPeer(newStore.getId());
```

Comment:
Remove this. Also consider previous suggestion that if you want to delegate everything to a cache miss and reload.